### PR TITLE
🐛 fix: Respect Immutable config for Body()

### LINKED ISSUE #3236: 🐛 [Bug]: Immutable configuration in Fiber not working as expected
### Bug Description

The `Immutable` configuration option in Fiber, when set to true, is intended to ensure that certain request values (e.g., request bodies) are immutable and accessible beyond the handler's lifecycle. However, enabling this option does not work as expected, resulting in data race errors when the request body is processed within a differernt goroutine.

Documentation Says:

> When set to true, this relinquishes the 0-allocation promise in certain cases in order to access the handler values (e.g., request bodies) in an immutable fashion so that these values are available even if you return from the handler.

Am I missing something here, or is this the expected behavior, and I need to manually copy the body to ensure immutability in my code?

### How to Reproduce

1. Create a Fiber app with the Immutable and StreamRequestBody configurations set to true
```go
app := fiber.New(fiber.Config{
    Immutable:         true,
    StreamRequestBody: true,
})
```

2. Define a route where the request body is processed in new goroutine:
```go
app.Get("/keep", func(c *fiber.Ctx) error {
    body := c.Body() // Supposed to be immutable
    go func() {
        for i := 0; i < 100; i++ {
            time.Sleep(time.Second)
            var data map[string]any
            err := json.Unmarshal(body, &data)
            fmt.Println(err, data) // Accessing body in a goroutine
        }
    }()
    return c.JSON(map[string]any{"success": true})
})
```

3. send multiple concurrent requests
```curl
curl --request GET 'http://localhost:8080/keep' \
--header 'Content-Type: application/json' \
--data-raw '{"key": "val"}'
```

### Expected Behavior

The request body `c.Body()` should remain immutable and safe to access across goroutines without causing data races but actually get a data races

```
==================
WARNING: DATA RACE
Read at 0x00c00011a080 by goroutine 76:
  encoding/json.checkValid()
  ...
Previous write at 0x00c00011a080 by goroutine 75:
  runtime.slicecopy()
  ...
==================
```

### Fiber Version

v2.52.5

### Code Snippet (optional)

```go
package main

import "github.com/gofiber/fiber/v3"
import "log"

func main() {
  app := fiber.New()

  // Steps to reproduce

  log.Fatal(app.Listen(":3000"))
}
```

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

### PR DESCRIPTION
**Pull request by nickajacks1**
# Description

For request bodies without encoding, the Immutable configuration was not being respected, leading to potential data races.
`go run -race` no longer produces warnings for the example in the original issue.

Fixes #3236 

## Type of change

Please delete options that are not relevant.

- [x] Enhancement (improvement to existing features and functionality)

## Checklist

Before you submit your pull request, please make sure you meet these requirements:

- [ ] Followed the inspiration of the Express.js framework for new functionalities, making them similar in usage.
- [ ] Conducted a self-review of the code and provided comments for complex or critical parts.
- [ ] Updated the documentation in the `/docs/` directory for [Fiber's documentation](https://docs.gofiber.io/).
- [ ] Added or updated unit tests to validate the effectiveness of the changes or new features.
- [ ] Ensured that new and existing unit tests pass locally with the changes.
- [ ] Verified that any new dependencies are essential and have been agreed upon by the maintainers/community.
- [ ] Aimed for optimal performance with minimal allocations in the new code.
- [ ] Provided benchmarks for the new code to analyze and improve upon.

## Commit formatting

Please use emojis in commit messages for an easy way to identify the purpose or intention of a commit. Check out the emoji cheatsheet here: [CONTRIBUTING.md](https://github.com/gofiber/fiber/blob/master/.github/CONTRIBUTING.md#pull-requests-or-commits)

### CODE REVIEW

**Comment by gaby** [comment]
@nickajacks1 This happens in v3 too right? If so can you send a PR for that target branch

File: ctx.go [hunk 1]
```diff
@@ -340,7 +340,11 @@ func (c *Ctx) Body() []byte {
 	// rule defined at: https://www.rfc-editor.org/rfc/rfc9110#section-8.4-5
 	encodingOrder = getSplicedStrList(headerEncoding, encodingOrder)
 	if len(encodingOrder) == 0 {
-		return c.fasthttp.Request.Body()
+		body = c.fasthttp.Request.Body()
```
Comment (gaby) [inline ctx.go:343]: Do we really have to set this variable? Why not

If immutable
  Return blah
else
  Return foo

**Comment by nickajacks1** [comment]
@gaby  no, v3 properly checks for immutability in the getBody() function.  I can update to add getBody.  Actually, we should probably just backport that function.  Let me make that update.

**Comment by gaby** [comment]
@nickajacks1 While you are at it, add go1.22 and 1.23 to the tests workflow

**Comment by nickajacks1** [comment]
Sure. This pr or separate?

**Comment by gaby** [comment]
> Sure. This pr or separate?

Same PR

**Review by gaby — APPROVED** [review]
👍 LGTM

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**